org.eclipse.xtext.ui.tests/src-longrunning/org/eclipse/xtext/ui/tests/editor/DirtyStateEditorSupportIntegrationTest.java

```diff
@@ -116,7 +123,62 @@ public void testUndoRedo() throws Exception {
 				.getQualifiedName().getLastSegment());
 		Iterables.getLast(events);
 	}
+	
+	/*
+	 * @see https://github.com/eclipse/xtext/issues/2385
+	 */
+	@Test public void testModifyFileInExternEditor() throws Exception {
+		IXtextDocument document = editor.getDocument();
+
+		Display.getDefault().readAndDispatch();
+		
+		assertNotEquals(document.get(), "");
+		
+		getWorkbench().getActiveWorkbenchWindow().getActivePage().getViewReferences()[0].getView(false).setFocus();
+		
+		File externalEditFile = ((IFileEditorInput)editor.getEditorInput()).getFile().getLocation().toFile();
+		try (FileWriter fw = new FileWriter(externalEditFile)) {
+			fw.write("");
+		}
+		editor.setFocus();
+		Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_REFRESH, null);
+		syncUtil.yieldToQueuedDisplayJobs(new NullProgressMonitor());
+		syncUtil.waitForReconciler(editor);
+		assertEquals(document.get(), "");
+	}
+
+	/**
+	 * This test needs a manual button click.
```

Comment:
This comment sounds strange: does a human have to click on the button to make the test proceed?

Reply:
Mine is not meant to be a review, but I don't understand what the test does: if it's ignored, it doesn't test anything.

I don't think clicking on a dialog button is (easily) doable in a plug-in test, but it's easy in a SWTBot test.

Maybe a SWTBot test to effectively verify the fix would be better.

But, again, I did not follow the whole bug and fix.

Reply:
Sure, therefore there is one automated test and one that needs manual interaction. 
Not sure if SWTBot is used in Xtext code.

Reply:
We use SWTBot tests, just a few, to test Xtend dialogs.

My main concern is that there's no automatic test for verifying the fix, again, if I understand correctly.

Reply:
There is one test which is executed automatically, which tests the basic functionality.. The ignored test is testing the case when the editor is dirty and the file is out of sync case.. I didn't saw an example how to automate a click on a button, without introducing a new dependency.

Reply:
@szarnekow : also the manual test is green without changes?

Reply:
For this the ignored test has to be unignored and executed manually..